docs/source/examples/dataframe-hdf5.rst

```diff
@@ -0,0 +1,146 @@
+
+.. code-block:: python
+   :emphasize-lines: 40,45
+   
+    
+    def create_hdf5s(iters):
+           
+       import random
+       import pandas as pd
+       import numpy as np
+       import string 
+       import json
+    
+       # Intitalize counter to build specific number of hdf5 files
+       count = 0
+       setseed=1
+       np.random.seed(seed=setseed)
+       random.seed(setseed)
+    
+       # Key to store group keys and file names for generated hdf5 files
+       fileKeys = {}
+    
+       while count <= iters:
+            
+           # randomly pick letter as dataset key
+           groupkey = random.choice(list(string.ascii_lowercase))
+        
+           # randomly pick a number as hdf5 filename
+           saver = 'my'+str(np.random.randint(100))+'.h5'
+        
+           # Make a dataframe; 26 rows, 2 columns
+           df = pd.DataFrame(
+           {'x': np.random.randint(1,1000,26),
+           'y': np.random.randint(1,1000,26)},
+           index=list(string.ascii_lowercase)
+           )
+        
+           # Write synthetic hdf5 to current directory
+           df.to_hdf(saver, key='/'+groupkey, format='table')
+           fileKeys[saver]=groupkey
+           count += 1
+           setseed+=1
+        
+       # Return the dictionary with file key and filename for testing
+       with open('hdf5Keys.json','w') as f:
+           f.write(json.dumps(fileKeys))
+
+    return fileKeys
+
+    # Create the HDF5 files and keys to access
+    hdf5Keys = create_hdf5s(keys)
+
+
+.. doctest:: 
+    
+    def create_hdf5s(iters):
+        import random
+        import pandas as pd
+        import numpy as np
+        import string 
+        import json
+
+        # Intitalize counter to build specific number of hdf5 files
+        count = 0
+
+        # Key to store group keys and file names for generated hdf5 files
+        fileKeys = {}
+
+        while count <= iters:
+            # randomly pick letter as dataset key
+            groupkey = random.choice(list(string.ascii_lowercase))
+
+            # randomly pick a number as hdf5 filename
+            saver = 'my'+str(np.random.randint(100))+'.h5'
+
+            # Make a dataframe; 26 rows, 2 columns
+            df = pd.DataFrame(
+            {'x': np.random.randint(1,1000,26),
+            'y': np.random.randint(1,1000,26)},
+            index=list(string.ascii_lowercase)
+            )
+
+            # Write synthetic hdf5 to current directory
+            df.to_hdf(saver, key='/'+groupkey, format='table')
+
+            fileKeys[saver]=groupkey
+
+            count += 1
+        # Return the dictionary with file key and filename for testing
+        with open('hdf5Keys.json','w') as f:
+            f.write(json.dumps(hdf5Keys))
+        return fileKeys
+    
+    # Build the exercise data
+    hdf5Keys = create_hdf5s(keys)
+        
+**Read single dataset from HDF5:**
```

Comment:
Thanks for helping out.  This is where my HDF5 knowledge is limited.  Are you referring to the ["Fixed Format" limitation in the pandas documentation for HDF5](
https://pandas-docs.github.io/pandas-docs-travis/io.html#fixed-format
)?  Here it is:

"The examples above show storing using put, which write the HDF5 to PyTables in a fixed array format, called the fixed format. These types of stores are are not appendable once written (though you can simply remove them and rewrite). Nor are they queryable; they must be retrieved in their entirety. They also do not support dataframes with non-unique column names. The fixed format stores offer very fast writing and slightly faster reading than table stores. This format is specified by default when using put or to_hdf or by format='fixed' or format='f' "